pynamodb/attributes.py

```diff
@@ -339,7 +339,7 @@ def _set_attributes(self, **attributes: Attribute) -> None:
                 raise ValueError("Attribute {} specified does not exist".format(attr_name))
             setattr(self, attr_name, attr_value)
 
-    def serialize(self, null_check=True) -> Dict[str, Dict[str, Any]]:
+    def _container_serialize(self, null_check: bool = True) -> Dict[str, Dict[str, Any]]:
```

Comment:
just ended up renaming it, since otherwise something like MapAttribute that derives from both Attribute and AttributeContainer ends up getting two incompatible `serialize` signatures